installation/setup-mlir.sh

```
@@ -91,22 +85,34 @@ RELEASES_JSON=$(curl -fL \
-if [ -z "$ASSETS_JSON" ] || [ "$ASSETS_JSON" = "null" ] || [ "$ASSETS_JSON" = "[]" ]; then
-  echo "Error: No release with LLVM $LLVM_VERSION found." >&2
+# Parse JSON to find matching release and extract download URLs
+# Use grep and sed for JSON parsing without external dependencies
+# This approach assumes GitHub API JSON format with "name" field followed by "browser_download_url"
+# The pattern is designed to work with the standard GitHub releases API response format
+# Compact JSON for easier processing
+RELEASES_JSON_COMPACT=$(echo "$RELEASES_JSON" | tr -d '\n' | sed 's/  */ /g')
+
+# Escape special regex characters in the match pattern to prevent regex interpretation
+# Escape these chars: . [ ] \ * ^ $ ( ) + ? { | }
+MATCH_PATTERN_ESCAPED=$(echo "$MATCH_PATTERN" | sed 's/[].\[\*^$()+?{|\\]/\\&/g')
```

Shellcheck (0.11.0)

[style] See if you can use ${variable//search/replace} instead.

(SC2001)